packages/analytics-browser/src/attribution/campaign-tracker.ts

```diff
@@ -0,0 +1,88 @@
+import { createIdentifyEvent, Identify } from '@amplitude/analytics-core';
+import {
+  Storage,
+  Campaign,
+  CampaignParser as ICampaignParser,
+  CampaignTracker as ICampaignTracker,
+  CampaignTrackerFunction,
+  CampaignTrackerOptions,
+} from '@amplitude/analytics-types';
+import { getCookieName as getStorageKey } from '../utils/cookie-name';
+import { CampaignParser } from './campaign-parser';
+import { EMPTY_VALUE } from './constants';
+
+export class CampaignTracker implements ICampaignTracker {
+  storageKey: string;
+  campaignParser: ICampaignParser;
+  tracker: CampaignTrackerFunction;
+  excludeReferrers: string[];
+  initialEmptyValue: string;
+  onNewCampaign?: (campaign: Campaign) => unknown;
+
+  constructor(public storage: Storage<Campaign>, options: CampaignTrackerOptions) {
+    this.campaignParser = new CampaignParser();
+    this.storageKey = getStorageKey(options.apiKey, 'MKTG');
+    this.tracker = options.tracker ?? (() => Promise.resolve());
+    this.excludeReferrers = options.excludeReferrers ?? [];
+    if (typeof location !== undefined) {
+      this.excludeReferrers.unshift(location.hostname);
+    }
+    this.initialEmptyValue = options.initialEmptyValue ?? EMPTY_VALUE;
+    this.onNewCampaign = options.onNewCampaign;
+  }
+
+  isNewCampaign(currentCampaign: Campaign, previousCampaign?: Campaign) {
```

Comment:
This is the logic I think we'll want to be able to parameterize maybe in a second iteration. Specifically, what if they just wanted to reset on new referrer but not new campaign Id?